src/mono/mono/metadata/marshal-ilgen.c

```diff
@@ -5231,14 +3980,6 @@ emit_marshal_vtype_ilgen (EmitMarshalContext *m, int argnum, MonoType *t,
 	return conv_arg;
 }
 
-static inline void
-emit_string_free_icall (MonoMethodBuilder *mb, MonoMarshalConv conv)
-{
-	if (conv == MONO_MARSHAL_CONV_BSTR_STR || conv == MONO_MARSHAL_CONV_ANSIBSTR_STR || conv == MONO_MARSHAL_CONV_TBSTR_STR)
-		mono_mb_emit_icall (mb, mono_free_bstr);
-	else
-		mono_mb_emit_icall (mb, mono_marshal_free);
-}
```

Comment:
@naricc I think you should leave `emit_string_free_icall` unshared - just use it in marshal-ilgen.c as is.

In the lightweight marshaler, you can implement a new  inline version of `emit_string_free_icall`:
```
static inline void
emit_string_free_icall (MonoMethodBuilder *mb, MonoMarshalConv conv)
{
	g_assert (conv == MONO_MARSHAL_CONV_NONE);
	mono_mb_emit_icall (mb, mono_marshal_free);
}
```

because lightweight marshaling won't do conversions in the runtime.